ibm/service/kms/data_source_ibm_kms_kmip_adapter.go

```diff
@@ -0,0 +1,108 @@
+// Copyright IBM Corp. 2017, 2021 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package kms
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceIBMKMSKmipAdapterBaseSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"adapter_id": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "The UUID of the KMIP adapter to be fetched",
+		},
+		"name": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "The name of the KMIP adapter to be fetched",
+		},
+		"description": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "The description of the KMIP adapter to be fetched",
+		},
+		"profile": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "The profile of the KMIP adapter to be fetched",
+		},
+		"profile_data": {
+			Type:        schema.TypeMap,
+			Computed:    true,
+			Description: "The data specific to the KMIP Adapter profile",
+		},
+		"created_by": &schema.Schema{
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "The unique identifier that is associated with the entity that created the adapter.",
+		},
+		"created_at": &schema.Schema{
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "The date when a resource was created. The date format follows RFC 3339.",
+		},
+		"updated_by": &schema.Schema{
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "The unique identifier that is associated with the entity that updated the adapter.",
+		},
+		"updated_at": &schema.Schema{
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "The date when a resource was updated. The date format follows RFC 3339.",
+		},
+	}
+}
+
+func DataSourceIBMKMSKmipAdapter() *schema.Resource {
```

Comment:
I do think this is a little less readable than just having those 3 attributes updated, but thats personal preference